collect_app/src/main/java/org/odk/collect/android/tasks/InstanceSyncTask.java

```diff
@@ -107,7 +107,9 @@ protected String doInBackground(Void... params) {
                     while (instanceCursor.moveToNext()) {
                         String instanceFilename = instanceCursor.getString(
                                 instanceCursor.getColumnIndex(InstanceColumns.INSTANCE_FILE_PATH));
-                        if (candidateInstances.contains(instanceFilename)) {
+                        String instanceStatus = instanceCursor.getString(
```

Comment:
@grzesiek2010 you're right. The sync used everything in the folder and we need to go through all records in the db to make sure we don't add duplicates and removed dangling records.